File: Algorithms/src/2144. Minimum Cost of Buying Candies With Discount/Code/solution.py
class Solution:
    def minimumCost(self, cost: List[int]) -> int:
        # Sort candies from highest cost to lowest cost
        cost.sort(reverse=True)

        total = 0

        # Every third candy becomes free
        for i in range(len(cost)):
            if i % 3 == 2:
                continue

            total += cost[i]

        return total

File: Algorithms/src/2144. Minimum Cost of Buying Candies With Discount/Code/solution.cpp
class Solution
{
public:
    int minimumCost(vector<int> &cost)
    {
        // Sort candies from highest cost to lowest cost
        sort(cost.begin(), cost.end(), greater<int>());

        int total = 0;

        // Add all candies except every third candy
        for (int i = 0; i < cost.size(); i++)
        {
            // Index 2,5,8,... are free candies
            if (i % 3 == 2)
                continue;

            total += cost[i];
        }

        return total;
    }
};